application/helpers/expressions/em_manager_helper.php

```diff
@@ -9800,6 +9752,86 @@ public static function htmlSpecialCharsUserValue($string)
         $string = str_replace(["{","}"], ["&#123;","&#125;"], $string);
         return $string;
     }
+
+    /**
+     * Save the starting values to the database
+     * @return boolean|void
+     */
+    public static function SaveStartingValues()
+    {
+        // Init
+        $LEM =& LimeExpressionManager::singleton();
+
+        // If not starting values, retun
+        if (!isset($_SESSION[$LEM->sessid]['startingValues'])) {
+            return false;
+        }
+        if (!is_array($_SESSION[$LEM->sessid]['startingValues'])) {
+            return false;
+        }
+
+        // set seed key if it doesn't exist to be able to pass count of startingValues check at next IF
+        // GJ: I think this is to assure that starting values always have a seed entry and then be able to check more easily
+        // If there are other starting values, besides the seed.
+        // I think this piece of code may never be reached by the execution thread. There is always a seed.
+        if (!array_key_exists('seed', $_SESSION[$LEM->sessid]['startingValues'])){        
+            $_SESSION[$LEM->sessid]['startingValues']['seed'] = '';
+        }
+
+        // Check if there are other starting values, besides the seed
+        // NOTE: now that we use a seed, count($_SESSION[$LEM->sessid]['startingValues']) start at 1
+        if (count($_SESSION[$LEM->sessid]['startingValues']) > 1) {
+            foreach ($_SESSION[$LEM->sessid]['startingValues'] as $k => $value) {
+                if (isset($LEM->knownVars[$k])) {
+                    $knownVar = $LEM->knownVars[$k];
+                } elseif (isset($LEM->qcode2sgqa[$k])) {
+                    $knownVar = $LEM->knownVars[$LEM->qcode2sgqa[$k]];
+                } else {
+                    continue;
+                }
+                if (!isset($knownVar['jsName'])) {
+                    continue;
+                }
+                switch ($knownVar['type']) {
+                    case Question::QT_D_DATE: //DATE
```

Comment:
Still needed ? Can not use only self::checkValidityAnswer ?

Reply:
Honestly, I don't recall that much.
This code is moved from here:https://github.com/LimeSurvey/LimeSurvey/pull/1754/files/50a5b0c212d5c1c9f041ad0f81224c007ced3108#diff-fa8d35264aa59c85d37ca2d1756472996499f06fb346865594378102f5f7abc0L4640

Don';t want to be rude, but don't think doing re-engineering right now (in the scope of this ticket) will help stability :)

From what I see, checkValidityAnswer don't have much for Dates.
I know the code is messy. Was like that.
Still, I think that if we go in the direction of re-engineering instead of re-organizing, we may create lot of unexpected bugs.

Reply:
Right , it's on todo :)

https://github.com/LimeSurvey/LimeSurvey/blob/3ff0af260ca43f09ad87a2ea3171a385e1e074dd/application/helpers/expressions/em_manager_helper.php#L9667-L9669